nixos/modules/system/etc/etc.nix

```diff
@@ -72,6 +87,30 @@ in
 
   options = {
 
+    system.etc.overlay = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Mount `/etc` as an overlayfs instead of generating it via a perl script.
+
+          Note: This is currently experimental. Only enable this option if you're
+          confident that you can recover your system if it breaks.
+        '';
+      };
+
+      mutable = mkOption {
+        type = types.bool;
+        default = true;
```

Comment:
If I set this to false, i must have configured /etc/machine-id otherwise strange things happen, right?

Reply:
Yes exactly, you would either create an empty /etc/machine-id via environment.etc and let systemd mount a tmpfs over it or you hard code it in your config (which might not be a bad idea). 

The immutable /etc is the one with the most unexpected things to happen. However since this is opt-in I think its fine for now.